pkg/executor/importer/import.go

```diff
@@ -1335,5 +1340,27 @@ func GetMsgFromBRError(err error) string {
 	return raw[:len(raw)-len(berrMsg)-len(": ")]
 }
 
+// GetTargetNodeCPUCnt get cpu count of target node where the import into job will be executed.
+// target node is current node if it's server-disk import or disttask is disabled,
+// else it's the node managed by disttask.
+// exported for testing.
+func GetTargetNodeCPUCnt(path string) (int, error) {
+	u, err2 := storage.ParseRawURL(path)
+	if err2 != nil {
+		return 0, exeerrors.ErrLoadDataInvalidURI.GenWithStackByArgs(plannercore.ImportIntoDataSource,
+			err2.Error())
+	}
+
+	serverDiskImport := storage.IsLocal(u)
+	if serverDiskImport || !variable.EnableDistTask.Load() {
+		return cpu.GetCPUCount(), nil
+	}
+	// the call path of initialization of threadCnt don't have context, so we use a timeout here.
+	ctx, cancel := context.WithTimeout(context.Background(), getCPUCountTimeout)
```

Comment:
What if have a heavy workload in CI？

Reply:
we can refactor the call path to pass a context, too lazy to change them, will think about it tomorrow

Reply:
https://github.com/pingcap/tidb/pull/50091/commits/058bb613a828b1ae87847fe412e3010f51f5af9f